@shopify/polaris-icons .svg files to use `currentColor`

alex-page opened the pull request:
### WHY are these changes introduced?

Our icon svg files set a specific fill value, this causes developers to copy paste icons from our library as they feel locked into the colors the icons provide.

This will have a follow up PR to update the `<Icon>` component to set a default color value and give the ability to use `currentColor` and stop any visual regressions.

### WHAT is this pull request doing?

Initial PR to change all icons to use `currentColor`.

alex-page committed:
Use currentColor for svg icons

alex-page committed:
Clean up SVG code

alex-page committed:
Merge branch 'polaris-icons-v8' into use-currentcolor

alex-page committed:
Revert icon changes in website code

alex-page committed:
Revert icon component changes

alex-page committed:
Update wild-ties-join.md

alex-page committed:
Merge branch 'polaris-icons-v8' into use-currentcolor

alex-page committed:
Revert icon component changes

alex-page committed:
Add new icons

sam-b-rose reviewed (approved):
Assuming we should merge this in before #11262 ?

alex-page committed:
Merge branch 'polaris-icons-v8' into use-currentcolor

alex-page committed:
Merge branch 'polaris-icons-v8' into use-currentcolor

alex-page committed:
Update new icons

alex-page committed:
Replace path fill with svg fill (#11262)

### WHY are these changes introduced?

`fill="currentColor"` is sometimes repeated 3-4 times depending on the
number of `<path>` elements in an `<svg>`. This PR moves
`fill="currentColor"` to the root `<svg>` element to simplify how the
color flows through the icon. The most complicated icons like
`SortAscending` and `SortDescending` which use `fill-opacity` still work
with the `fill` value moved to the parent. I also searched for `circle`
and `polygon` elements but they were unused.

### WHAT is this pull request doing?

Reducing the size of our SVG icons by moving the `fill` property to the
`<svg>`

alex-page committed:
Fix tests, add missing fill and remove clip-rule
